src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/AllelePseudoDepth.java

```diff
@@ -0,0 +1,170 @@
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.tools.walkers.mutect.SomaticLikelihoodsEngine;
+import org.broadinstitute.hellbender.tools.walkers.mutect.SubsettedLikelihoodMatrix;
+import org.broadinstitute.hellbender.utils.MathUtils;
+import org.broadinstitute.hellbender.utils.genotyper.AlleleLikelihoods;
+import org.broadinstitute.hellbender.utils.genotyper.LikelihoodMatrix;
+import org.broadinstitute.hellbender.utils.help.HelpConstants;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.text.DecimalFormat;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@DocumentedFeature(groupName= HelpConstants.DOC_CAT_ANNOTATORS, groupSummary=HelpConstants.DOC_CAT_ANNOTATORS_SUMMARY, summary="Total depth of coverage per sample and over all samples (DP)")
+public final class AllelePseudoDepth implements GenotypeAnnotation {
+
+    private static DecimalFormat DEPTH_FORMAT = new DecimalFormat("#.##");
+    private static DecimalFormat FRACTION_FORMAT = new DecimalFormat("#.####");
+
+    public static final String DEPTH_KEY = "DD";
+    public static final String FRACTION_KEY = "DF";
+
+    public static final VCFFormatHeaderLine DEPTH_HEADER_LINE =
+            new VCFFormatHeaderLine(DEPTH_KEY, VCFHeaderLineCount.R, VCFHeaderLineType.Float, "Allele depth based on Dirchlet posterior pseudo-counts");
+    public static final VCFFormatHeaderLine FRACTION_HEADER_LINE =
+            new VCFFormatHeaderLine(FRACTION_KEY, VCFHeaderLineCount.R, VCFHeaderLineType.Float, "Allele Fraction based on Dirchilet postrior pseudo-counts");
+
+    @Argument(fullName = "dirichlet-prior-pseudo-count",
+              doc = "Pseudo-count used as prior for all alleles. The default is 1.0 resulting in a flat prior")
+    @Advanced
+    public double prior = 1.0;
+
+    @Argument(fullName = "dirichlet-keep-prior-in-count",
+              doc = "By default we don't keep the prior use in the output counts ase it makes it easier to interpret" +
+                      "this quanity as the number of supporting reads specially in low depth sites. We this toggled the prior is included")
+    public boolean keepPriorInCount = false;
+
+    @Argument(fullName = "pseudo-count-weight-decay-rate",
+              doc = "A what rate the weight of a read decreases base on its informativeness; e.g. 1.0 is " +
+                      "linear decay (default), 2.0 is for quadratic decay",
+              minValue = 0.0)
+    public double weightDecay = 1.0;
+
+    private static final List<String> KEYS = Arrays.asList(DEPTH_KEY, FRACTION_KEY);
+    private static final List<VCFCompoundHeaderLine> HEADER_LINES = Arrays.asList(DEPTH_HEADER_LINE, FRACTION_HEADER_LINE);
+
+    private double[][] priorPseudoCounts = new double[10][]; // possibly won't ever go beyond 10.
```

Comment:
This seems really dangerous, what's the justification for 10?

Reply:
we resize this array as needed, 10 is magic number that would cover 99% of cases (10 alleles or less).
I can give it a name.